stats.py

```diff
@@ -78,26 +90,84 @@ def passive_skill_tree() -> tuple[dict, dict]:
 		if 'masteryEffects' not in node:
 			continue
 		for effect in node['masteryEffects']:
-			masteries_dict[effect['effect']] = effect['stats']
+			masteries_dict[effect['effect']] = {"name": node["name"], "stats": effect['stats']}
 	return tree_dict, masteries_dict
 
 matchers = [(re.compile(pattern), attr) for pattern, attr in [
 	(r'\+(\d+) to maximum Life', 'flat_life'),
 	(r'(\d+)% increased maximum Life', 'increased_life'),
 	(r'\+(\d+) to (Strength.*|all Attributes)', 'strength'),
 	(r'(\d+)% increased effect of Non-Curse Auras from your Skills$', 'aura_effect'),
+	(r'(.*) has (\d+)% increased Aura Effect', 'specific_aura_effect'),
+	(r'(.\d+) to Level of all (.*) Gems', 'global_level'),
+	(r'(.\d+)% to Quality of all (.*) Gems', 'global_quality'),
+	(r'Allocates (.*)', 'additional_notable')
 ]]
 
 def _parse_item(stats: Stats, item: dict):
-	for modlist in ['implicitMods', 'explicitMods', 'craftedMods']:
+	for modlist in ['implicitMods', 'explicitMods', 'craftedMods', 'enchantMods']:
 		if modlist not in item:
 			continue
 		_parse_mods(stats, item[modlist])
 
 def _parse_mods(stats: Stats, mods: list) -> None:
 	for mod in mods:
 		for regex, attr in matchers:
-			m = regex.match(mod)
+			m = regex.search(mod)
 			if m:
+				if attr == "specific_aura_effect":
+					aura_name = m.group(1)
+					if aura_name in stats.specific_aura_effect:
+						stats.specific_aura_effect[aura_name] += int(m.group(2))
+					else:
+						stats.specific_aura_effect[aura_name] = int(m.group(2))
+					continue
+
+				if attr == "global_level":
+					stats.global_gem_level_increase += parse_gem_descriptor(m.group(2).lower(), int(m.group(1)))
+					continue
+
+				if attr == "global_quality":
+					stats.global_gem_quality_increase += parse_gem_descriptor(m.group(2).lower(), int(m.group(1)))
+					continue
+
+				if attr == "additional_notable":
+					notable = m.group(1)
+					if " if you have the matching modifier on" in notable:
+						notable = notable.split(" if you have the matching modifier on")[0]
+					stats.additional_notables |= {hash_for_notable(notable)}
+					continue
+
 				value = int(m.group(1))
 				setattr(stats, attr, getattr(stats, attr) + value)
+
+def parse_gem_descriptor(descriptor: str, value: int) -> list[tuple[list, int]]:
```

Comment:
since you always lowercase the descriptor before passing it in, maybe lowercase it in the function instead